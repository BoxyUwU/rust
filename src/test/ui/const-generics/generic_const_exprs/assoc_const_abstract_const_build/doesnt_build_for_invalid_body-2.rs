// this shouldnt compile even though the returned value of
// `<() as Trait<N>>::ASSOC` is always `N` which should unify with `N`

#![feature(generic_const_exprs)]
#![allow(incomplete_features)]

trait Trait<const N: usize> {
    const ASSOC: usize;
}

impl<const N: usize> Trait<N> for () {
    const ASSOC: usize = {
        let b = |a: String| todo!();
        N
    };
}

fn foo<const N: usize>() -> [(); <() as Trait<N>>::ASSOC] {
    [(); N]
    //~^ error: mismatched types
}

fn main() {}
