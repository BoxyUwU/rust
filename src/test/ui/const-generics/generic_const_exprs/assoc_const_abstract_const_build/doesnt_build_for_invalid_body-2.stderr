error[E0308]: mismatched types
  --> $DIR/doesnt_build_for_invalid_body-2.rs:19:5
   |
LL |     [(); N]
   |     ^^^^^^^ expected `<() as Trait<N>>::ASSOC`, found `N`
   |
   = note: expected type `<() as Trait<N>>::ASSOC`
              found type `N`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
